### PR TITLE
feat(types): Add support for X/Twitter v2 OAuth provider

### PR DESCRIPTION
**Pull request by kostaspt**
## Description

Add required types for X/Twitter v2.

image

## Checklist

- [X] `npm test` runs as expected.
- [X] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [X] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [X] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [X] `@clerk/clerk-js`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [X] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

**Comment by kostaspt** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @kostaspt - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 1.0.1-snapshot.v71dda29 |
| @clerk/chrome-extension | 1.0.1-snapshot.v71dda29 |
| @clerk/clerk-js | 5.0.1-snapshot.v71dda29 |
| @clerk/clerk-expo | 1.0.1-snapshot.v71dda29 |
| @clerk/fastify | 1.0.1-snapshot.v71dda29 |
| gatsby-plugin-clerk | 5.0.1-snapshot.v71dda29 |
| @clerk/localizations | 2.0.1-snapshot.v71dda29 |
| @clerk/nextjs | 5.0.1-snapshot.v71dda29 |
| @clerk/clerk-react | 5.0.1-snapshot.v71dda29 |
| @clerk/remix | 4.0.1-snapshot.v71dda29 |
| @clerk/clerk-sdk-node | 5.0.1-snapshot.v71dda29 |
| @clerk/shared | 2.0.1-snapshot.v71dda29 |
| @clerk/themes | 2.0.1-snapshot.v71dda29 |
| @clerk/types | 4.0.1-snapshot.v71dda29 |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/backend`
```sh
npm i @clerk/backend@1.0.1-snapshot.v71dda29 --save-exact
```
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@1.0.1-snapshot.v71dda29 --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.0.1-snapshot.v71dda29 --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@1.0.1-snapshot.v71dda29 --save-exact
```
`@clerk/fastify`
```sh
npm i @clerk/fastify@1.0.1-snapshot.v71dda29 --save-exact
```
`gatsby-plugin-clerk`
```sh
npm i gatsby-plugin-clerk@5.0.1-snapshot.v71dda29 --save-exact
```
`@clerk/localizations`
```sh
npm i @clerk/localizations@2.0.1-snapshot.v71dda29 --save-exact
```
`@clerk/nextjs`
```sh
npm i @clerk/nextjs@5.0.1-snapshot.v71dda29 --save-exact
```
`@clerk/clerk-react`
```sh
npm i @clerk/clerk-react@5.0.1-snapshot.v71dda29 --save-exact
```
`@clerk/remix`
```sh
npm i @clerk/remix@4.0.1-snapshot.v71dda29 --save-exact
```
`@clerk/clerk-sdk-node`
```sh
npm i @clerk/clerk-sdk-node@5.0.1-snapshot.v71dda29 --save-exact
```
`@clerk/shared`
```sh
npm i @clerk/shared@2.0.1-snapshot.v71dda29 --save-exact
```
`@clerk/themes`
```sh
npm i @clerk/themes@2.0.1-snapshot.v71dda29 --save-exact
```
`@clerk/types`
```sh
npm i @clerk/types@4.0.1-snapshot.v71dda29 --save-exact
```

**Comment by kostaspt** [comment]
!snapshot

**Comment by kostaspt** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @kostaspt - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 1.0.1-snapshot.v5bebd29 |
| @clerk/chrome-extension | 1.0.1-snapshot.v5bebd29 |
| @clerk/clerk-js | 5.0.1-snapshot.v5bebd29 |
| @clerk/clerk-expo | 1.0.1-snapshot.v5bebd29 |
| @clerk/fastify | 1.0.1-snapshot.v5bebd29 |
| gatsby-plugin-clerk | 5.0.1-snapshot.v5bebd29 |
| @clerk/localizations | 2.0.1-snapshot.v5bebd29 |
| @clerk/nextjs | 5.0.1-snapshot.v5bebd29 |
| @clerk/clerk-react | 5.0.1-snapshot.v5bebd29 |
| @clerk/remix | 4.0.1-snapshot.v5bebd29 |
| @clerk/clerk-sdk-node | 5.0.1-snapshot.v5bebd29 |
| @clerk/shared | 2.0.1-snapshot.v5bebd29 |
| @clerk/themes | 2.0.1-snapshot.v5bebd29 |
| @clerk/types | 4.0.1-snapshot.v5bebd29 |

Tip: Use the snippet copy button below to quickly install the required packages.
`@clerk/backend`
```sh
npm i @clerk/backend@1.0.1-snapshot.v5bebd29 --save-exact
```
`@clerk/chrome-extension`
```sh
npm i @clerk/chrome-extension@1.0.1-snapshot.v5bebd29 --save-exact
```
`@clerk/clerk-js`
```sh
npm i @clerk/clerk-js@5.0.1-snapshot.v5bebd29 --save-exact
```
`@clerk/clerk-expo`
```sh
npm i @clerk/clerk-expo@1.0.1-snapshot.v5bebd29 --save-exact
```
`@clerk/fastify`
```sh
npm i @clerk/fastify@1.0.1-snapshot.v5bebd29 --save-exact
```
`gatsby-plugin-clerk`
```sh
npm i gatsby-plugin-clerk@5.0.1-snapshot.v5bebd29 --save-exact
```
`@clerk/localizations`
```sh
npm i @clerk/localizations@2.0.1-snapshot.v5bebd29 --save-exact
```
`@clerk/nextjs`
```sh
npm i @clerk/nextjs@5.0.1-snapshot.v5bebd29 --save-exact
```
`@clerk/clerk-react`
```sh
npm i @clerk/clerk-react@5.0.1-snapshot.v5bebd29 --save-exact
```
`@clerk/remix`
```sh
npm i @clerk/remix@4.0.1-snapshot.v5bebd29 --save-exact
```
`@clerk/clerk-sdk-node`
```sh
npm i @clerk/clerk-sdk-node@5.0.1-snapshot.v5bebd29 --save-exact
```
`@clerk/shared`
```sh
npm i @clerk/shared@2.0.1-snapshot.v5bebd29 --save-exact
```
`@clerk/themes`
```sh
npm i @clerk/themes@2.0.1-snapshot.v5bebd29 --save-exact
```
`@clerk/types`
```sh
npm i @clerk/types@4.0.1-snapshot.v5bebd29 --save-exact
```

**Review by chanioxaris — APPROVED** [review]
💯

**Review by dimkl — APPROVED** [review]

File: .changeset/smooth-numbers-shout.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+'@clerk/clerk-js': patch
+'@clerk/backend': patch
+'@clerk/types': patch
+---
+
+Add support for X/Twitter v2 OAuth provider
```
Comment (dimkl) [inline .changeset/smooth-numbers-shout.md:7]: 🙃 i would expect it to be a minor version change.

**Review by dimkl — APPROVED** [review]

**Pull request merged**